aspose/src/main/java/org/frankframework/extensions/aspose/services/conv/impl/convertors/MailConvertor.java

```diff
@@ -90,6 +97,7 @@ class MailConvertor extends AbstractConvertor {
 	protected MailConvertor(CisConversionService cisConversionService, CisConfiguration configuration) {
 		super(configuration, MEDIA_TYPE_LOAD_FORMAT_MAPPING.keySet());
 		this.cisConversionService = cisConversionService;
+		this.attachmentConversionExecutor = Executors.newFixedThreadPool(configuration.getMaxThreads());
```

Comment:
I was wondering if it would be a good idea to use here a https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ForkJoinPool.html and in the `convert()` method use a parallelStream() with this custom ForkJoinPool to do the multi-threaded conversion and collection of results?

See also:
https://www.baeldung.com/java-8-parallel-streams-custom-threadpool

Reply:
The problem is the type of `Attachment` currently does not support `.stream()` methods.

Although this problem could be solved by changing the underlying base type of `Attachment`.

Reply:
Ah, I see.

Well, it was just a suggestion. If it won't work, then no problem.

Reply:
I mean `AttachmentCollection`, not `Attachment` itself..

But since these are `Aspose` classes,  it's a bit hard to solve it.

Reply:
Actually I found you could do this:

`StreamSupport.stream(attachments.spliterator(), true)`

Reply:
> Actually I found you could do this:
> 
> `StreamSupport.stream(attachments.spliterator(), true)`

Will investigate in the weeknd into this !

Reply:
Ive tried something but I dont think this is much better